File: parallel-programming/avg-run-test.cs
using System;

class AvgRunTest
{

    static int n = 4096;
    static int[,] a = new int[n, n];

    static void Main()
    {
        Random r = new Random();        
        // randomizing our initial problem
        for(int row = 0; col < n; row++){
            for(int col = 0; col < n; col++){
                a[row][col] = r.Next(1, 10);
            }
        }

        // starting out timer
        var watch = System.Diagnostics.Stopwatch.StartNew();
        
        // running avg for each array
        double totalAvg = 0.0;
        for(int row = 0; col < n; row++){
            double localAvg = 0.0;
            for(int col = 0; col < n; col++){
                localAvg += col;
            }
            localAvg = localAvg / n;
            totalAvg += localAvg;
        }

        // getting final results
        watch.Stop();
        double elasped = watch.ElaspsedMilliseconds;
        totalAvg = totalAvg / n;

        // the total_avg is a debug function to check that the randomized values within range of expected
        System.Console.WriteLine("The loop took: " + str(elasped) + " ms and the avg is: " + str(totalAvg));
    }
}

File: parallel-programming/avg-run-test.py
import random, time

n = 4096

# randomizing our initial problem
a = [[random.randint(1,10) for x in range(n)] for y in range(n)]

# starting out timer
start = time.time()

# running avg for each array
total_avg = 0.0
for row in a:
    local_avg = 0.0
    for col in row:
        local_avg += col
    local_avg = local_avg / n
    total_avg += local_avg

# getting final results
end = time.time()
elasped = end - start
total_avg = total_avg / n

# the total_avg is a debug function to check that the randomized values within range of expected
print("The loop took: " + str(elasped) + ' ms and the avg is: ' + str(total_avg))